src/common/src/metrics.rs

```diff
@@ -44,3 +48,49 @@ impl Atomic for TrAdderAtomic {
 }
 
 pub type TrAdderGauge = GenericGauge<TrAdderAtomic>;
+
+pub struct CustomLayer {
+    pub aws_sdk_retry_counts: GenericCounter<AtomicU64>,
+    pub aws_http_timeout_retry_counts: GenericCounter<AtomicU64>,
+}
+
+impl<S> Layer<S> for CustomLayer
+where
+    S: Subscriber + for<'a> LookupSpan<'a>,
+{
+    fn on_event(&self, event: &tracing::Event<'_>, _ctx: Context<'_, S>) {
+        if event.metadata().target() == "aws_smithy_client::retry"
+            && event.metadata().level() == &tracing::Level::DEBUG
```

Comment:
This is too hacky because there can be[ multiple debug logs](https://docs.rs/aws-smithy-client/0.55.3/src/aws_smithy_client/retry.rs.html#357) emitted in aws_smithy_client::retry, which can make the metric incorrect. We need to be more specific here.

Reply:
How about checking the level first and then the target? We can short-circuit the string comparison in common caess I guess.